jib-core/src/integration-test/java/com/google/cloud/tools/jib/api/ContainerizerIntegrationTest.java

```diff
@@ -218,7 +219,60 @@ public void testSteps_forBuildToDockerRegistry_multipleTags()
   }
 
   @Test
-  public void tesBuildToDockerRegistry_dockerHubBaseImage()
+  public void testSteps_forBuildToDockerRegistry_skipExistingDigest()
+      throws IOException, InterruptedException, ExecutionException, RegistryException,
+          CacheDirectoryCreationException {
+    System.setProperty(JibSystemProperties.SKIP_EXISTING_IMAGES, "true");
+
+    JibContainer image1 =
+        buildRegistryImage(
+            ImageReference.of("gcr.io", "distroless/java", DISTROLESS_DIGEST),
+            ImageReference.of("localhost:5000", "testimagerepo", "testtag"),
+            Collections.singletonList("testtag2"));
+
+    // Test that the initial image with the original tag has been pushed.
+    String imageReference = "localhost:5000/testimagerepo:testtag";
+    localRegistry.pull(imageReference);
+    assertDockerInspect(imageReference);
+    Assert.assertEquals(
+        "Hello, world. An argument.\n", new Command("docker", "run", "--rm", imageReference).run());
+
+    // Test that any additional tags have also been pushed with the original image.
+    String imageReference2 = "localhost:5000/testimagerepo:testtag2";
+    localRegistry.pull(imageReference2);
+    assertDockerInspect(imageReference2);
+    Assert.assertEquals(
+        "Hello, world. An argument.\n",
+        new Command("docker", "run", "--rm", imageReference2).run());
+
+    // Push the same image with a different tag, with SKIP_EXISTING_IMAGES enabled.
+    JibContainer image2 =
+        buildRegistryImage(
+            ImageReference.of("gcr.io", "distroless/java", DISTROLESS_DIGEST),
+            ImageReference.of("localhost:5000", "testimagerepo", "new_testtag"),
+            Collections.emptyList());
+
+    // Test that the pull request throws an exception, indicating that the new tag was not pushed.
+    String imageReference3 = "localhost:5000/testimagerepo:new_testtag";
+    try {
+      localRegistry.pull(imageReference3);
+      Assert.fail(
+          "jib.skipExistingImages was enabled and digest was already pushed, "
+              + "hence testtag2 shouldn't have been pushed.");
+    } catch (RuntimeException ignore) {
+      // As expected, registry throws exception that manifest is unknown.
```

Comment:
Please catch the specific exception and verify the exception message (`assertEquals(..., ex.getMessage())` or `assertThat(ex.getMessage(), ....containsString(...))` for example).

Reply:
The exception thrown is a `RuntimeException`. I'll add assertions on the message though.

Reply:
Sounds good. Please verify message.

Reply:
nit: per our style, it should be `ex` instead of `e`:

```java
 } catch (RuntimeException ex) {
```